falcon/status_codes.py

```diff
@@ -115,6 +127,14 @@
 HTTP_GATEWAY_TIMEOUT = HTTP_504
 HTTP_505 = '505 HTTP Version not supported'
 HTTP_HTTP_VERSION_NOT_SUPPORTED = HTTP_505
+HTTP_506 = '506 Variant Also Negotiates'
+HTTP_VARIANT_ALSO_NEGOTIATES = HTTP_506
+HTTP_507 = '507 Insufficient Storage'
+HTTP_INSUFFICIENT_STORAGE = HTTP_507
+HTTP_508 = '508 Loop Detected'
+HTTP_LOOP_DETECTED = HTTP_508
+HTTP_510 = '510 Not Extended'
```

Comment:
I'm hesitant to include 510, since it is also from a very old experimental RFC to define an extension mechanism for the HTTP protocol itself.